haystack/nodes/retriever/dense.py

```diff
@@ -1087,15 +1068,6 @@ def retrieve_batch(
         if batch_size is None:
             batch_size = self.batch_size
 
-        if isinstance(filters, list):
-            if len(filters) != len(queries):
-                raise HaystackError(
-                    "Number of filters does not match number of queries. Please provide as many filters"
-                    " as queries or a single filter that will be applied to each query."
-                )
```

Comment:
Moved down to `BaseDocumentStore.query_by_embedding_batch` and `SearchEngineDocumentStore.query_by_embedding_batch